eqt/ui/UIFormWidget.py

```diff
@@ -41,58 +41,199 @@ def createForm(self):
-        except KeyError:
-            logging.info('Widget ' + name + ' does not have a label.')
+        self.insertWidgetToFormLayout(-1, name, qwidget, qlabel)
+
+    def _addToWidgetDictionary(self, dictionary, name, qwidget, qlabel = None):
+        '''Adds the field, and label if present, in the widget dictionary.'''
+        dictionary[f'{name}_field'] = qwidget
+        if qlabel is not None:
+            dictionary[f'{name}_label'] = qlabel
+        
+    def _addToWidgetNumberDictionary(self, name, widget_number): 
+        '''
+        Adds one item in the widget-number dictionary whose key is name and value is 
+        the current widget number (i.e. row) in the form layout.
+        As one widget is inserted, the widget numbers associated to the other widgets 
+        in the layout are updated.
+
+        Parameters:
+        ---------------
+        name: string
+            name of the widget
+        widget_number : int
+            position of the widget in the form layout, i.e. row, in the current state
+        '''
+        if widget_number == -1:
+            self.widget_number_dictionary[name] = self.num_widgets - 1 
+        else:
+            for key, value in self.widget_number_dictionary.items():
+                if value >= widget_number:
+                    self.widget_number_dictionary[key] = value + 1
+            self.widget_number_dictionary[name] = widget_number
+
+    def _popWidgetNumberDictionary(self, name):
+        '''
+        Removes one item in the widget-number dictionary whose key is name.
+        As one widget is removed, the widget numbers associated to the other widgets 
+        in the layout are updated.
+
+        Parameters:
+        ---------------
+        name: string
+            name of the widget
+        '''
+        widget_number = self.getWidgetNumber(name)
+        for key, value in self.widget_number_dictionary.items():
+            if value > widget_number:
+                self.widget_number_dictionary[key] = value - 1
+        self.widget_number_dictionary.pop(name) 
 
+    def _popWidgetFromDictionary(self, dictionary, name):
+        '''
+        Removes the item(s) associated with `name` from a dictionary.
+
+        Parameters:
+        -----------------
+        dictionary :  dict
+        name: str
+            Format: {name}
+        '''
+        if name in dictionary.keys():
+            dictionary.pop(name) 
+        if f'{name}_field' in dictionary.keys():
+            dictionary.pop(f'{name}_field')      
+        if f'{name}_label' in dictionary.keys():
+            dictionary.pop(f'{name}_label') 
+
+    def addWidget(self, qwidget, qlabel, name):
+        '''
+        Adds a widget and a label widget, or a spanning widget, at the the end of
+        the form layout. 
+
+        Parameters:
+        ----------
+        name: str
+        qwidget: widget
+        qlabel: qlabel widget or str
+        '''
+        self._addWidget(name, qwidget, qlabel)
```

Comment:
I would remove `_addWidget` and call instead `insertWidgetToFormLayout` and possibly remove totally `_addWidget`

Reply:
I removed _addwidget